docs/extensions/civix.md

```diff
@@ -1 +1,636 @@
-Coming soon!
+# Civix
+
+>> ![information](../img/info.png)
+Note: This page started
+[here in the wiki](https://wiki.civicrm.org/confluence/display/CRMDOC/Create+a+Module+Extension).
+We have not yet transferred all of the script snipets, so you may wish to
+peruse this page as well.
+
+## Introduction
+
+The [`civix`](https://github.com/totten/civix/) command-line tool is the
+community-endorsed method for building your CiviCRM extensions.
+
+Follow the installation instructions in the
+[GitHub repository](https://github.com/totten/civix/).
+
+In order to verify that all your configuration is correct ping CiviCRM from
+within your extensions directory with:
+
+```bash
+civix civicrm:ping
+```
+
+This command should reply with "Ping successful".
+If the ping is unsuccessful, re-read the civix README.md and do the
+post-installation configuration.
+
+It is also useful to examine the output of the `civix help` command to read
+about what `civix` can do for you.
+
+Help is available on individual commands, e.g.:
+
+```bash
+civix help civicrm:ping
+```
+
+## Generating a skeletal extension
+
+To generate a skeletal extension module, we will use `civix generate:module`
+and pass in the name for our extension. See [here](./basics/#extension-names)
+for details of naming conventions.
+
+Start with:
+
+```bash
+cd $YOUR_EXTENSIONS_DIR
+civix generate:module --help
+```
+
+Then use a command like this:
+
+```bash
+civix generate:module com.example.myextension --license=AGPL-3.0
+```
+
+This command will report that it has created three files, following the
+[standard extension structure](./files).
+
+The command attempts to autodetect authorship information (your name and
+email address) by reading your
+[`git`](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)
+configuration. If this fails or is
+otherwise incorrect, then you may pass explicit values with `--author`
+and `--email`.
+
+You can now update your `info.xml`. This file initially contains some
+examples and placeholders which you need to fix. You can edit most of these
+fields intuitively.  If you need detailed specifications, see [Extension
+Reference](http://wiki.civicrm.org/confluence/display/CRMDOC/Extension+Reference).
+
+Now that you have created your extension, you need to activate it by navigating
+to:
+
+**Administer » System Settings » Manage Extensions**
+
+or
+
+**Administer » Customize Data and Screens » Manage Extensions.**
+
+For more detailed instructions, see
+[Extensions](http://wiki.civicrm.org/confluence/display/CRMDOC/Extensions).
+
+## Add features
+
+There are many different features that you can add to a module-extension
+at your discretion. A few possibilities:
+
+### Add a basic web page
+
+CiviCRM uses a typical web-MVC architecture. To implement a basic web
+page, you must create a PHP controller class, create a Smarty template
+file, and create a routing rule. You can create the appropriate files by
+calling "civix generate:page"
```

Comment:
For inline references to code/commands/symbols, you can use backtick. Compare: "civix generate:page" vs `civix generate:page`